spinta/config/__init__.py

```diff
@@ -74,115 +73,106 @@
-
-def update_config_from_cli(config, options):
-    options = (o.split('=', 1) for o in options)
-    options = ((k.split('.'), v) for k, v in options)
-    return update_config(config, options)
-
-
-def update_config_from_env(config, env):
-    options = get_options_from_env(CONFIG, env)
-    for path, template in TEMPLATES.items():
-        name = '_'.join(['SPINTA'] + [p.upper() for p in path])
-        if name in env:
-            for key in env[name].split(','):
-                _config = get_by_path(CONFIG, path + (template,))
-                options += get_options_from_env(_config, env, path + (key,))
-    return update_config(config, options)
-
-
-def update_config_from_env_file(config, path: str):
-    env = {}
-    with pathlib.Path(path).open() as f:
-        for line in f:
-            if line.startswith('#'):
-                continue
-            line = line.strip()
-            if line == '':
-                continue
-            if '=' not in line:
-                continue
-            name, value = line.split('=', 1)
-            env[name] = value
-    return update_config_from_env(config, env)
-
-
-def get_options_from_env(config, env, base=()):
-    options = []
-    for path, v in traverse(config):
-        name = '_'.join(['SPINTA'] + [p.upper() for p in base + path])
-        if name in env:
-            options.append((base + path, env[name]))
-    return options
-
-
-def get_by_path(config, path):
-    this = config
-    for key in path:
-        if isinstance(this, dict) and key in this:
-            this = this[key]
-        else:
-            path = '.'.join(path)
-            raise Exception(f"Unknow configuration parameter {path!r}.")
-    return this
-
-
-def get_copy_by_path(config, path):
-    return copy.deepcopy(get_by_path(config, path))
-
-
-def traverse(value, path=()):
+    def keys(self, *key, **kwargs):
+        return self.get(*key, cast=list, **kwargs)
+
+    def _get_config_value(self, key: tuple, default, env):
+
+        # 1. Get value from comman line arguments.
+        if key in self._config['cliargs']:
+            return self._config['cliargs'][key]
+
+        # Auto-generate environment variable name.
+        if env is True:
+            env = 'SPINTA_' + '_'.join(key).upper()
+
+        # 2. Get value from environment.
+        if env and env in self._config['environ']:
+            return self._config['environ'][env]
+
+        # 3. Get value from env file.
+        if env and env in self._config['envfile']:
+            return self._config['envfile'][env]
+
+        # 4. Get value from default configs.
+        if key in self._config['default']:
+            return self._config['default'][key]
+
+        return default
+
+
+def _traverse(value, path=()):
     if isinstance(value, dict):
         for k, v in value.items():
-            yield from traverse(v, path + (k,))
+            yield from _traverse(v, path + (k,))
     else:
         yield path, value
+
+
+def _get_inner_keys(value):
```

Comment:
As far as I understand `value` is of type `dict`. Which makes it very confusing when we work with dictionary's keys and _values_.

In my opinion, better names could be `map_value`, `dict_value`, `map_`, `dict_`, etc.